pkg/kubernetes/kubernetes.js

```diff
@@ -225,5 +236,123 @@ define([
         return new EtcdClient();
     };
 
+    kubernetes.discover = function discover(address, callback) {
```

Comment:
Why was this added. It is not present in git master. I would suggest basing your branch from git master. I'd be willing to help you rebase this if you like.